### PR TITLE
fix: wait for backing image restoration finished before checking volume restoration

### PR DESCRIPTION
**Pull request by ChanYiLin**
ref: https://github.com/longhorn/longhorn/issues/8515

We should check if BackingImage is restored and before checking volume is restored.
Because Volume needs to wait until BackingImage to restored.
It may timeout checking the status , because Volume is not finished yet.

### CODE REVIEW

**Comment by ChanYiLin** [comment]
## Testing...

File: manager/integration/tests/test_system_backup_restore.py [hunk 1]
```diff
@@ -118,7 +123,7 @@ def test_system_backup_and_restore_volume_with_data(client, volume_name, set_ran
     check_volume_data(restored_volume, data)
 
 @pytest.mark.system_backup_restore   # NOQA
-def test_system_backup_and_restore_volume_with_backingimage(client, volume_name, set_random_backupstore):  # NOQA
+def test_system_backup_and_restore_volume_with_backingimage(client, core_api, volume_name, set_random_backupstore):  # NOQA
```
Comment (yangchiu) [inline manager/integration/tests/test_system_backup_restore.py:126]: The test case can't be run. Got error:
```
_ ERROR at setup of test_system_backup_and_restore_volume_with_backingimage[s3] _
file /integration/tests/test_system_backup_restore.py, line 125
  @pytest.mark.system_backup_restore   # NOQA
  def test_system_backup_and_restore_volume_with_backingimage(client, core_api, volume_name, set_random_backupstore):  # NOQA
E       fixture 'core_api' not found
```

**Review by c3y1huang — DISMISSED** [review]

**Review by yangchiu — CHANGES_REQUESTED** [review]

File: manager/integration/tests/test_system_backup_restore.py (hunk 1 above)
Comment (ChanYiLin) [reply manager/integration/tests/test_system_backup_restore.py:127]: fixed, thanks

**Comment by ChanYiLin** [comment]
Hi @yangchiu 
I have fixed the comment
While testing, I found there is a bug which make the test flaky
I have created the PR to fix it: https://github.com/longhorn/longhorn-manager/pull/2823

You can wait for the fix PR to be merged, and test it and merge it after.

**Comment by yangchiu** [comment]
Merge this first to reveal the issue of https://github.com/longhorn/longhorn-manager/pull/2823.

**Comment by ChanYiLin** [comment]
Hi @yangchiu 
We can first verified the system restore with BackingImage in this issue and close the issue first:　https://github.com/longhorn/longhorn/issues/8515


Then for e2e flaky issue because of backup failed 
The fix has changed to this implementation: https://github.com/longhorn/longhorn-manager/pull/2835
We can use this issue to verified: https://github.com/longhorn/longhorn/issues/8650

Thanks!

**Pull request merged**